### PR TITLE
update Resource spec based on the move to the SDK and named tracers

### PR DESCRIPTION
**Pull request by tsloughter**
The Resource spec was moved from the API to the SDK but it still referred to the API in the text. This also updates to reference the Providers and all the tracers that may be retrieved from the provider (named tracers and meters).

### CODE REVIEW

File: specification/sdk-resource.md [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
 decoupling of discovery of resource information from exporters. This allows for
 independent development and easy customization for users that need to integrate
-with closed source environments. API MUST allow for creation of `Resources` and
+with closed source environments. SDK MUST allow for creation of `Resources` and
 for associating them with telemetry.
 
 When used with distributed tracing, a resource can be associated with the
-[TracerSdk](sdk-tracing.md#tracer-sdk). When associated with `TracerSdk`, all
-`Span`s produced by the `Tracer`, that is implemented by this `TracerSdk`, will
+[TracerProvider](sdk-tracing.md#tracer-sdk). When associated with
```
Comment (MrAlias) [inline specification/sdk-resource.md]:
```suggestion
[TracerProvider](sdk-tracing.md#tracer-sdk). When associated with a
```

File: specification/sdk-resource.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+`TracerProvider` all `Span`s produced by any `Tracer` from the provider will
```
Comment (MrAlias) [inline specification/sdk-resource.md]:
```suggestion
`TracerProvider`, all `Span`s produced by any `Tracer` from the provider will
```

File: specification/sdk-resource.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
 automatically be associated with this `Resource`.
 
 When used with metrics, a resource can be associated with the
-[MeterSdk](sdk-metrics.md#meter-sdk). When associated with `MeterSdk`, all
-`Metrics` produced by this `Meter`, that is implemented by this `MeterSdk`, will
-automatically be associated with this `Resource`.
+[MeterProvider](sdk-metrics.md#meter-sdk). When associated with `MeterProvider`
```
Comment (MrAlias) [inline specification/sdk-resource.md]:
```suggestion
[MeterProvider](sdk-metrics.md#meter-sdk). When associated with a `MeterProvider`,
```

File: specification/sdk-resource.md [hunk 4]
```diff
@@ -1,25 +1,25 @@
 # Resource SDK
 
 A [Resource](overview.md#resources) represents the entity producing telemetry.
```
Comment (dyladan) [inline specification/sdk-resource.md]: I don't like the language "the entity producing telemetry". Is this the telemetry library? the end user application? the library being monitored? Maybe an example like "represents the entity being monitored, such as an end user application or service".

Same issue I brought up on https://github.com/open-telemetry/opentelemetry-specification/pull/422, the `Factory => Provider` rename OTEP is not yet merged and maybe we should wait for that?

Comment (tsloughter) [reply specification/sdk-resource.md]: I can try to reword it.

Comment (dyladan) [reply specification/sdk-resource.md]: Even just an example would be very helpful.

Comment (tsloughter) [reply specification/sdk-resource.md]: Ok, I've added an example, does this help?

File: specification/sdk-resource.md [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
+telemetry. For example, a process producing telemetry that is running in a
+container on Kubernetes has a Pod name, it is in a namespace and possibly is
+part of a Deployment which also has a name. All three of these attributes can be
+included in the `Resource`.
+
+The primary purpose of resources as a first-class concept in the SDK is
 decoupling of discovery of resource information from exporters. This allows for
 independent development and easy customization for users that need to integrate
-with closed source environments. API MUST allow for creation of `Resources` and
+with closed source environments. SDK MUST allow for creation of `Resources` and
```
Comment (jkwatson) [inline specification/sdk-resource.md]: Can we say "The SDK", rather than just "SDK"?

Comment (jkwatson) [inline specification/sdk-resource.md]:
```suggestion
with closed source environments. The SDK MUST allow for creation of `Resources` and
```

**Pull request merged**